bin/fetchl10nstrings.js

```diff
@@ -0,0 +1,341 @@
+ * Create the string bundles and then save them to disk.
+ */
+/*jshint node:true, browser:false, esnext:true*/
+"use strict";
+const async = require("marcosc-async");
+const fs = require("fs");
+const path = require("path");
+const clc = require("cli-color");
+const fetch = require("node-fetch"); // jshint ignore:line
+const repo = "https://hg.mozilla.org/releases/l10n/mozilla-aurora/";
+const newTabPath = "/raw-file/tip/browser/chrome/browser/newTab";
+const globalDirPath = "/raw-file/tip/dom/chrome/global.dtd";
+const l10nPath = path.resolve(`${__dirname}/../l10n/`);
+const defaultLocale = {};
+
+// CLI-Colors
+const error = clc.red.bold;
+const warn = clc.yellow;
+const notice = clc.blue;
+
+function fetchAndProcessTask(processor) {
+  return async(function*(url) {
+    let response = yield assureResponse(yield fetch(url));
+    let text = yield response.text();
+    return processor(text);
+  });
+}
+
+const dtdProcessorTask = fetchAndProcessTask(processDTD);
+const propProcessorsTask = fetchAndProcessTask(processProps);
+
+/**
+ * Helper object for representing string bundles.
+ *
+ * @constructor
+ * @param {String} locale The locale for the string bundle.
+ */
+function StringBundle(locale) {
+  this.locale = locale;
+}
+
+StringBundle.prototype = {
+  get properties() {
+    return `${repo}${this.locale}${newTabPath}.properties`;
+  },
+  get dtd() {
+    return `${repo}${this.locale}${newTabPath}.dtd`;
+  },
+  get dir() {
+    return `${repo}${this.locale}${globalDirPath}`;
+  },
+  /**
+   * Saves the bundle to disk.
+   *
+   * @return {Promise} Resolves once writing to disk is done is done.
+   */
+  save() {
+    return async.task(function*() {
+      // kick off downloads
+      let results = yield Promise.all([
+        dtdProcessorTask(this.dir),
+        dtdProcessorTask(this.dtd),
+        propProcessorsTask(this.properties)
+      ]);
+      validateResults(results, this.locale);
+      // Reduce resulting objects into a single object, using the defaultLocale
+      // as the base.
+      let dirtyObject = results.reduce(
+        (prev, next) => Object.assign(prev, next), Object.assign({}, defaultLocale)
+      );
+      try {
+        let trimmedObject = trimRedundantProps.call(this, dirtyObject);
+        let sortedObject = makeSortedObject(trimmedObject);
+        let text = JSON.stringify(sortedObject, null, 2);
+        yield writeToDisk(text, this.locale);
+      } catch (err) {
+        console.error(error(err));
+      }
+    }, this);
+  }
+};
+/**
+ * As object's property-order is not guaranteed in JS, we need to sort them
+ * so we can see what's actually changed. Sort is natural as per ES.
+ *
+ * @param  {Object} unsortedObject The object whose properties are not sorted.
+ * @return {Object} A new Object that has been sorted.
+ */
+function makeSortedObject(unsortedObject) {
+  return Object.getOwnPropertyNames(unsortedObject)
+    .sort()
+    .reduce((obj, next) => {
+      obj[next] = unsortedObject[next];
```

Comment:
Actually, they don't serialize fine :( They serialize to `{}` with no properties - so that would just lead us back here. I'm pretty sure the order is guaranteed tho... I think things go funny when properties get deleted tho (hence me needing to do this).